buildtest/config.py

```diff
@@ -108,33 +111,43 @@ def load_configuration(config_path=None):
 # Run on init, so we only load once
 config_opts = load_configuration()
 
-supported_launcher_dict = {
-    "local": ["local", "mpirun", "mpiexec"],
-    "slurm": ["mpirun", "mpiexec", "srun"]
-}
 
-def validate_queues(config_queues)
+def validate_queues(queues):
```

Comment:
These validation functions are defeating the purpose of having the jsonschema package - it's like going back to the previous code that had to manually check all these fields, etc. We want the validation to happen with the jsonschema, not via hard coded if/else statements.

Reply:
i understand, i had some trouble redesigning the schema again, if i were to do it again, I would and by no means this PR is perfect. I didn't want to break my existing work so happy to get second revision

Reply:
I think implementing something complex that would need to be removed is not the right way to address this. If you need help with the schema, then you should ask and I can put it on my queue to take a look. But adding and merging something that will be removed because it isn't what we would consider the best solution? That doesn't make sense, and it's not good use of your time.

Reply:
I dont anticipate to get the schema correct the first time, It was a learning experience so not a complete waste of time. 

What i had in mind was having ability to define one or more local scheduler followed by other workload managers (slurm). The way i was thinking was something along the lines of this.

```
{
    "editor": "vi",
    "queues": 
     {
       "local": 
        {
          "localhost": {                      
            "description": "submit job locally",            
            "launcher": "local"
            "vars:" {
              "FOO": "BAR"
            }
            "modules": {
              "purge": true
              "name": ["GCC"]                
            }
              
          },
          "local_mpirun": {
            "description": "submit job locally via mpirun",
            "launcher": "mpirun",            
          }
        }
        "slurm": {
          "admin": {
              "description": "admin partition",
              "launcher": "srun",
              "options": [
                "#SBATCH -n 1"
                ]
          }
        }
     }
}
```

Reply:
can you explain the choice of localhost? localhost usually refers to 127.0.0.1 which can be confusing in this context.

Reply:
That's probably a bad name convention. The ``localhost``  and ``local_mpirun`` are queues of type ``local`` which means they are meant to run tests locally without a scheduler. Note that grouping queues by `local` allows me to get rid of ``scheduler: local`` and i can now have unique keys that represent local queues.  

Currently, buildtest supports running test locally, the idea was to allow test-configuration reference one or more queues to run job through a scheduler which could be ``local`` or an actual scheduler. I also had in mind a ``ssh`` launcher which would run test using ``ssh`` on ``localhost`` but it could be configured to use another host. This assumes passwordless-ssh and is kinda out-of scope of this PR. The only value i see with ``ssh`` is ability to run job in a clean environment on any node for user that don't have a scheduler. 

There were few others things i had in mind which were not addressed
- It could be nice to allow user to specify path to launcher types, or buildtest can auto-detect  which is what it is doing now
- Currently, i am checking that slurm queue exists with one defined in configuration file, i think that should be changed by allowing a ``partition`` key and the name (``admin`` in example above) could be arbitrary. Reason is a user could want more than queue configuration using one partition without getting a name conflict in json. Hope that makes sense.

We got few options we could do:
1. If you have a proposed buildtest site configuration (Stanford) you would like to use please share your proposal and I  can try to implement the schema and update this PR
2. I can temporarily close the PR, work on my branch until this is complete and reissue PR for review, though we will lose all the reviews so you would have to do it again.
3. We can PR merge as is or with a few minor modification if you have any. 
4. You can branch from my PR and merge into mine, or re-issue PR to upstream. 

Let me know what your thoughts.

Reply:
Let me take a shot sometime this week at playing around with the schema, I think I understand what you are potentially trying to do. I definitely don’t want to lose all the work you’ve done thus far for this PR, it’s good and we just need this final detail.

Reply:
Sounds good, in that case i can keep my branch and PR open for time being. I kinda like your proposal of introducing json to validate schema using **jsonschema**, makes life easy in terms of schema validation and allow for more elaborate configurations. I am planning on start developing a site configuration for stampede2 with some tests in a repo, i think practically using buildtest at a site will help build and refine the features. It's bit hard to do this at my local cluster.